src/en/mangamo/src/eu/kanade/tachiyomi/extension/en/mangamo/Mangamo.kt

```diff
@@ -0,0 +1,435 @@
+        }
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val dto = response.body.string().parseJson<DocumentDto<SeriesDto>>().fields
+        return processSeries(dto)
+    }
+
+    // Chapter list section
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val uri = manga.url.toHttpUrl()
+
+        val seriesId = uri.queryParameter(MangamoConstants.SERIES_QUERY_PARAM)!!.toInt()
+
+        val seriesObservable = client.newCall(
+            firestore.getDocument("Series/$seriesId") {
+                fields = listOf(
+                    SeriesDto::maxFreeChapterNumber.name,
+                    SeriesDto::onlyTransactional.name,
+                )
+            },
+        ).asObservableSuccess().map { response ->
+            response.body.string().parseJson<DocumentDto<SeriesDto>>().fields
+        }
+
+        val chaptersObservable = client.newCall(
+            firestore.getCollection("Series/$seriesId/chapters") {
+                fields = listOf(
+                    ChapterDto::enabled.name,
+                    ChapterDto::id.name,
+                    ChapterDto::seriesId.name,
+                    ChapterDto::chapterNumber.name,
+                    ChapterDto::name.name,
+                    ChapterDto::createdAt.name,
+                    ChapterDto::onlyTransactional.name,
+                )
+
+                orderBy = listOf(descending(ChapterDto::chapterNumber.name))
+            },
+        ).asObservableSuccess().map { response ->
+            response.body.string().parseJson<QueryResultDto<ChapterDto>>().elements
+        }
+
+        val hideCoinChapters = coinMangaPref.contains(MangamoConstants.HIDE_COIN_MANGA_OPTION_CHAPTERS)
+
+        return Observable.combineLatest(seriesObservable, chaptersObservable) { series, chapters ->
+            chapters
+                .mapNotNull { chapter ->
+                    if (chapter.enabled != true) {
+                        return@mapNotNull null
+                    }
+
+                    val isFreeChapter = chapter.chapterNumber!! <= (series.maxFreeChapterNumber ?: 0)
+
+                    val isCoinChapter = chapter.onlyTransactional == true ||
+                        (series.onlyTransactional == true && !isFreeChapter)
+
+                    if (hideCoinChapters && isCoinChapter) {
+                        return@mapNotNull null
+                    }
+
+                    SChapter.create().apply {
+                        chapter_number = chapter.chapterNumber
+                        date_upload = chapter.createdAt!!
+                        name = chapter.name +
+                            if (isFreeChapter) {
+                                " (Free)"
+                            } else if (isCoinChapter) {
+                                " (Coin)"
+                            } else {
+                                ""
+                            }
```

Comment:
imo, add a lock emoji on paid chapters and/or coin empji on coin chapters; free chapters shouldn't have any indicator